scripts/source-maintenance.sh

```diff
@@ -92,13 +106,13 @@ if ! git diff --quiet; then
 	exit 1
 fi
 
-# On squid-cache.org we have to use the python scripted md5sum
-HOST=`hostname`
-if test "$HOST" = "squid-cache.org" ; then
-	MD5="md5"
-else
-	MD5="md5sum"
-fi
+for MD5 in "${MD5:-md5}" md5sum shasum sha1sum
```

Comment:
I have the impression we are overcomplicating things here.
The checksum algorithm here only has one use, detect if astyle broke a file while changing it. Any file that has md5-like capabilities works for the purpose. We don't need to be cryptographically secure, and the result is ephemeral and it will not be distributed.
I'll add a specific test, but I have the feeling we're creeping scope a lot here.
The current implementation only works with md5sum in practice, nobody is complaining, nor passing custom checksum scripts. Someone might theoretically do it in the future but if they do then they can also edit the script

Reply:
> We don't need to be cryptographically secure, and the result is ephemeral and it will not be distributed.

Very true but completely irrelevant to my change request AFAICT.

> I'll add a specific test, but I have the feeling we're creeping scope a lot here.

You may be right, but, again, the scope creep (if any) is unrelated to my change request.

* The PR adds a way to manually specify a checksum program. Whether you want to call that a scope creep is up to you.
* The PR adds a way to automatically find a working checksum program. Whether you want to call that a scope creep is up to you.
* The PR checks whether the manually specified checksum programs works. Whether you want to call that a scope creep is up to you.
* The change request essentially asks not to mix the two mutually exclusive things together. That is a PR bug fix, not a scope creep! If you _want_ to check whether the manually specified checksum program is working, then do it correctly -- the script has to exit if the manually configured program does not work. However, the PR does not  _have to_ check the manually specified program -- removing the third bullet from the PR scope/code is an acceptable solution IMO.

Reply:
I have added an explicit check. Had I not done that, I would have taken the option to remove the check altogether